```
2026-06-23T10:48:21.9462964Z [ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.15.0:compile (default-compile) on project quarkus-maven-plugin: Compilation failure: Compilation failure: 
2026-06-23T10:48:21.9465602Z [ERROR] /home/runner/_work/quarkus/quarkus/devtools/maven/src/main/java/io/quarkus/maven/CreateExtensionMojo.java:[7,35] package org.fusesource.jansi does not exist
2026-06-23T10:48:21.9467867Z [ERROR] /home/runner/_work/quarkus/quarkus/devtools/maven/src/main/java/io/quarkus/maven/CreateExtensionMojo.java:[7,1] static import only from classes and interfaces
2026-06-23T10:48:21.9470557Z [ERROR] /home/runner/_work/quarkus/quarkus/devtools/maven/src/main/java/io/quarkus/maven/CreateJBangMojo.java:[5,35] package org.fusesource.jansi does not exist
2026-06-23T10:48:21.9472617Z [ERROR] /home/runner/_work/quarkus/quarkus/devtools/maven/src/main/java/io/quarkus/maven/CreateJBangMojo.java:[5,1] static import only from classes and interfaces
2026-06-23T10:48:21.9474584Z [ERROR] /home/runner/_work/quarkus/quarkus/devtools/maven/src/main/java/io/quarkus/maven/CreateProjectMojo.java:[4,35] package org.fusesource.jansi does not exist
2026-06-23T10:48:21.9476490Z [ERROR] /home/runner/_work/quarkus/quarkus/devtools/maven/src/main/java/io/quarkus/maven/CreateProjectMojo.java:[4,1] static import only from classes and interfaces
2026-06-23T10:48:21.9478520Z [ERROR] /home/runner/_work/quarkus/quarkus/devtools/maven/src/main/java/io/quarkus/maven/CreateProjectMojo.java:[33,28] package org.fusesource.jansi does not exist
2026-06-23T10:48:21.9480426Z [ERROR] /home/runner/_work/quarkus/quarkus/devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java:[91,37] package org.fusesource.jansi.internal does not exist
2026-06-23T10:48:21.9482511Z [ERROR] /home/runner/_work/quarkus/quarkus/devtools/maven/src/main/java/io/quarkus/maven/CreateExtensionMojo.java:[248,31] cannot find symbol
2026-06-23T10:48:21.9493509Z [ERROR]   symbol:   method ansi()
2026-06-23T10:48:21.9494300Z [ERROR]   location: class io.quarkus.maven.CreateExtensionMojo
2026-06-23T10:48:21.9495654Z [ERROR] /home/runner/_work/quarkus/quarkus/devtools/maven/src/main/java/io/quarkus/maven/CreateJBangMojo.java:[137,27] cannot find symbol
2026-06-23T10:48:21.9496883Z [ERROR]   symbol:   method ansi()
2026-06-23T10:48:21.9497612Z [ERROR]   location: class io.quarkus.maven.CreateJBangMojo
2026-06-23T10:48:21.9499131Z [ERROR] /home/runner/_work/quarkus/quarkus/devtools/maven/src/main/java/io/quarkus/maven/CreateProjectMojo.java:[489,17] cannot find symbol
2026-06-23T10:48:21.9500368Z [ERROR]   symbol:   method ansi()
2026-06-23T10:48:21.9501067Z [ERROR]   location: class io.quarkus.maven.CreateProjectMojo
2026-06-23T10:48:21.9502406Z [ERROR] /home/runner/_work/quarkus/quarkus/devtools/maven/src/main/java/io/quarkus/maven/CreateProjectMojo.java:[492,25] package Ansi does not exist
2026-06-23T10:48:21.9504231Z [ERROR] /home/runner/_work/quarkus/quarkus/devtools/maven/src/main/java/io/quarkus/maven/CreateProjectMojo.java:[490,23] cannot find symbol
2026-06-23T10:48:21.9508461Z [ERROR]   symbol:   method ansi()
2026-06-23T10:48:21.9509651Z [ERROR]   location: class io.quarkus.maven.CreateProjectMojo
2026-06-23T10:48:21.9511080Z [ERROR] /home/runner/_work/quarkus/quarkus/devtools/maven/src/main/java/io/quarkus/maven/CreateProjectMojo.java:[497,83] package Ansi does not exist
2026-06-23T10:48:21.9512905Z [ERROR] /home/runner/_work/quarkus/quarkus/devtools/maven/src/main/java/io/quarkus/maven/CreateProjectMojo.java:[497,17] cannot find symbol
2026-06-23T10:48:21.9514109Z [ERROR]   symbol:   method ansi()
2026-06-23T10:48:21.9514821Z [ERROR]   location: class io.quarkus.maven.CreateProjectMojo
2026-06-23T10:48:21.9516248Z [ERROR] /home/runner/_work/quarkus/quarkus/devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java:[627,55] cannot find symbol
2026-06-23T10:48:21.9517404Z [ERROR]   symbol:   variable Kernel32
2026-06-23T10:48:21.9518045Z [ERROR]   location: class io.quarkus.maven.DevMojo
2026-06-23T10:48:21.9519710Z [ERROR] /home/runner/_work/quarkus/quarkus/devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java:[627,33] cannot find symbol
2026-06-23T10:48:21.9521263Z [ERROR]   symbol:   variable Kernel32
2026-06-23T10:48:21.9529161Z [ERROR]   location: class io.quarkus.maven.DevMojo
2026-06-23T10:48:21.9533929Z [ERROR] /home/runner/_work/quarkus/quarkus/devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java:[628,40] cannot find symbol
2026-06-23T10:48:21.9535103Z [ERROR]   symbol:   variable Kernel32
2026-06-23T10:48:21.9535784Z [ERROR]   location: class io.quarkus.maven.DevMojo
2026-06-23T10:48:21.9537135Z [ERROR] /home/runner/_work/quarkus/quarkus/devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java:[630,41] cannot find symbol
2026-06-23T10:48:21.9538703Z [ERROR]   symbol:   variable Kernel32
2026-06-23T10:48:21.9539432Z [ERROR]   location: class io.quarkus.maven.DevMojo
2026-06-23T10:48:21.9540608Z [ERROR] /home/runner/_work/quarkus/quarkus/devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java:[634,25] cannot find symbol
2026-06-23T10:48:21.9552568Z [ERROR]   symbol:   variable Kernel32
2026-06-23T10:48:21.9553644Z [ERROR]   location: class io.quarkus.maven.DevMojo
2026-06-23T10:48:21.9554839Z [ERROR] /home/runner/_work/quarkus/quarkus/devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java:[655,51] cannot find symbol
2026-06-23T10:48:21.9555977Z [ERROR]   symbol:   variable Kernel32
2026-06-23T10:48:21.9556719Z [ERROR]   location: class io.quarkus.maven.DevMojo
2026-06-23T10:48:21.9557985Z [ERROR] /home/runner/_work/quarkus/quarkus/devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java:[655,29] cannot find symbol
2026-06-23T10:48:21.9559385Z [ERROR]   symbol:   variable Kernel32
2026-06-23T10:48:21.9560007Z [ERROR]   location: class io.quarkus.maven.DevMojo
2026-06-23T10:48:21.9561347Z [ERROR] /home/runner/_work/quarkus/quarkus/devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java:[656,36] cannot find symbol
2026-06-23T10:48:21.9562537Z [ERROR]   symbol:   variable Kernel32
2026-06-23T10:48:21.9563221Z [ERROR]   location: class io.quarkus.maven.DevMojo
2026-06-23T10:48:21.9564393Z [ERROR] /home/runner/_work/quarkus/quarkus/devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java:[657,17] cannot find symbol
2026-06-23T10:48:21.9570296Z [ERROR]   symbol:   variable Kernel32
2026-06-23T10:48:21.9571016Z [ERROR]   location: class io.quarkus.maven.DevMojo
2026-06-23T10:48:21.9571691Z [ERROR] -> [Help 1]
2026-06-23T10:48:21.9573192Z org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.15.0:compile (default-compile) on project quarkus-maven-plugin: Compilation failure
```

--- a/bom/application/pom.xml
+++ b/bom/application/pom.xml
@@ -154,7 +154,7 @@
         <kotlin-serialization.version>1.10.0</kotlin-serialization.version>
         <kotlin-metadata.version>0.9.0</kotlin-metadata.version>
         <azure.toolkit-lib.version>0.53.0</azure.toolkit-lib.version>
-        <dekorate.version>4.1.5</dekorate.version> <!-- Please check with Java Operator SDK team before updating -->
+        <dekorate.version>4.1.9</dekorate.version> <!-- Please check with Java Operator SDK team before updating -->
         <maven-invoker.version>3.2.0</maven-invoker.version>
         <awaitility.version>4.3.0</awaitility.version>
         <jboss-logmanager.version>3.2.1.Final</jboss-logmanager.version>
@@ -199,8 +199,8 @@
         <opensearch-testcontainers.version>2.1.3</opensearch-testcontainers.version>
         <com.dajudge.kindcontainer>2.0.0</com.dajudge.kindcontainer>
         <aesh.version>2.8.2</aesh.version>
-        <aesh-readline.version>2.6</aesh-readline.version>
-        <jansi.version>2.4.0</jansi.version> <!-- Keep in sync with aesh-readline and dekorate -->
+        <aesh-readline.version>2.6.3</aesh-readline.version>
+        <jline.version>4.2.1</jline.version> <!-- Keep in sync with aesh-readline and dekorate -->
         <!-- these two artifacts needs to be compatible together -->
         <strimzi-oauth.version>0.16.1</strimzi-oauth.version>
         <strimzi-oauth.nimbus.version>10.7</strimzi-oauth.nimbus.version>
@@ -6410,9 +6410,9 @@
                 <version>${aesh-readline.version}</version>
             </dependency>
             <dependency>
-                <groupId>org.fusesource.jansi</groupId>
+                <groupId>org.jline</groupId>
                 <artifactId>jansi</artifactId>
-                <version>${jansi.version}</version>
+                <version>${jline.version}</version>
             </dependency>
             <dependency>
                 <groupId>org.aesh</groupId>

--- a/core/deployment/src/main/java/io/quarkus/deployment/dev/testing/HtmlAnsiOutputStream.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/dev/testing/HtmlAnsiOutputStream.java
@@ -4,10 +4,10 @@ import java.io.IOException;
 import java.io.OutputStream;
 import java.nio.charset.StandardCharsets;
 
-import org.fusesource.jansi.AnsiColors;
-import org.fusesource.jansi.AnsiMode;
-import org.fusesource.jansi.AnsiType;
-import org.fusesource.jansi.io.AnsiOutputStream;
+import org.jline.jansi.AnsiColors;
+import org.jline.jansi.AnsiMode;
+import org.jline.jansi.AnsiType;
+import org.jline.jansi.io.AnsiOutputStream;
 
 public class HtmlAnsiOutputStream extends AnsiOutputStream {
 
@@ -17,7 +17,8 @@ public class HtmlAnsiOutputStream extends AnsiOutputStream {
     private static final byte[] BYTES_GT = "&gt;".getBytes();
 
     public HtmlAnsiOutputStream(OutputStream os) {
-        super(os, new ZeroWidthSupplier(), AnsiMode.Force, new HtmlAnsiProcessor(os), AnsiType.Redirected, AnsiColors.Colors256,
+        super(os, new ZeroWidthSupplier(), AnsiMode.Force, new HtmlAnsiProcessor(os),
+                AnsiType.Redirected, AnsiColors.Colors256,
                 StandardCharsets.UTF_8, null, null, true);
     }
 

--- a/core/deployment/src/main/java/io/quarkus/deployment/dev/testing/HtmlAnsiProcessor.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/dev/testing/HtmlAnsiProcessor.java
@@ -5,7 +5,7 @@ import java.io.OutputStream;
 import java.util.ArrayList;
 import java.util.List;
 
-import org.fusesource.jansi.io.AnsiProcessor;
+import org.jline.jansi.io.AnsiProcessor;
 
 public class HtmlAnsiProcessor extends AnsiProcessor {
 

--- a/core/deployment/src/main/java/io/quarkus/deployment/steps/NativeImageConfigBuildStep.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/steps/NativeImageConfigBuildStep.java
@@ -19,6 +19,7 @@ import io.quarkus.deployment.builditem.nativeimage.NativeImageProxyDefinitionBui
 import io.quarkus.deployment.builditem.nativeimage.NativeImageResourceBundleBuildItem;
 import io.quarkus.deployment.builditem.nativeimage.NativeImageSystemPropertyBuildItem;
 import io.quarkus.deployment.builditem.nativeimage.RuntimeInitializedClassBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.RuntimeInitializedPackageBuildItem;
 import io.quarkus.deployment.pkg.NativeConfig;
 import io.quarkus.deployment.pkg.steps.NativeOrNativeSourcesBuild;
 import io.quarkus.runtime.ssl.SslContextConfigurationRecorder;
@@ -86,6 +87,13 @@ class NativeImageConfigBuildStep {
         // runtime
         runtimeReInitClass.produce(new RuntimeInitializedClassBuildItem("org.wildfly.common.net.HostName"));
         runtimeReInitClass.produce(new RuntimeInitializedClassBuildItem("io.smallrye.common.net.HostName"));
+    }
+
+    @BuildStep(onlyIf = NativeOrNativeSourcesBuild.class)
+    RuntimeInitializedPackageBuildItem jlineNativRuntimeInit() {
+        // org.jline.nativ classes use JNI native methods for terminal interaction;
+        // their static initializers load native libraries which aren't available at build time
+        return new RuntimeInitializedPackageBuildItem("org.jline.nativ");
     }
 
     private Boolean isSslNativeEnabled(SslNativeConfigBuildItem sslNativeConfig,

--- a/devtools/maven/pom.xml
+++ b/devtools/maven/pom.xml
@@ -240,7 +240,7 @@
                         <configuration>
                             <artifactSet>
                                 <includes>
-                                    <include>org.fusesource.jansi:jansi</include>
+                                    <include>org.jline:jansi</include>
                                 </includes>
                             </artifactSet>
                         </configuration>

--- a/devtools/maven/src/main/java/io/quarkus/maven/CreateExtensionMojo.java
+++ b/devtools/maven/src/main/java/io/quarkus/maven/CreateExtensionMojo.java
@@ -4,7 +4,7 @@ import static io.quarkus.devtools.commands.CreateExtension.extractQuarkiverseExt
 import static io.quarkus.devtools.commands.CreateExtension.isQuarkiverseGroupId;
 import static org.apache.commons.lang3.StringUtils.isBlank;
 import static org.apache.commons.lang3.StringUtils.isEmpty;
-import static org.fusesource.jansi.Ansi.ansi;
+import static org.jline.jansi.Ansi.ansi;
 
 import java.io.File;
 import java.io.IOException;

--- a/devtools/maven/src/main/java/io/quarkus/maven/CreateJBangMojo.java
+++ b/devtools/maven/src/main/java/io/quarkus/maven/CreateJBangMojo.java
@@ -2,7 +2,7 @@ package io.quarkus.maven;
 
 import static io.quarkus.devtools.commands.CreateJBangProject.CreateJBangProjectKey.NO_JBANG_WRAPPER;
 import static io.quarkus.devtools.project.CodestartResourceLoadersBuilder.codestartLoadersBuilder;
-import static org.fusesource.jansi.Ansi.ansi;
+import static org.jline.jansi.Ansi.ansi;
 
 import java.io.File;
 import java.io.IOException;

--- a/devtools/maven/src/main/java/io/quarkus/maven/CreateProjectMojo.java
+++ b/devtools/maven/src/main/java/io/quarkus/maven/CreateProjectMojo.java
@@ -1,7 +1,7 @@
 package io.quarkus.maven;
 
 import static io.quarkus.devtools.project.CodestartResourceLoadersBuilder.codestartLoadersBuilder;
-import static org.fusesource.jansi.Ansi.ansi;
+import static org.jline.jansi.Ansi.ansi;
 
 import java.io.File;
 import java.io.IOException;
@@ -30,7 +30,7 @@ import org.eclipse.aether.RepositorySystem;
 import org.eclipse.aether.RepositorySystemSession;
 import org.eclipse.aether.impl.RemoteRepositoryManager;
 import org.eclipse.aether.repository.RemoteRepository;
-import org.fusesource.jansi.Ansi;
+import org.jline.jansi.Ansi;
 
 import io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver;
 import io.quarkus.devtools.commands.CreateProject;

--- a/devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java
+++ b/devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java
@@ -88,7 +88,7 @@ import org.eclipse.aether.resolution.DependencyRequest;
 import org.eclipse.aether.resolution.DependencyResolutionException;
 import org.eclipse.aether.resolution.DependencyResult;
 import org.eclipse.aether.util.artifact.JavaScopes;
-import org.fusesource.jansi.internal.Kernel32;
+import org.jline.nativ.Kernel32;
 
 import io.quarkus.bootstrap.BootstrapConstants;
 import io.quarkus.bootstrap.app.ApplicationModelSerializer;

--- a/extensions/picocli/deployment/src/main/java/io/quarkus/picocli/deployment/PicocliProcessor.java
+++ b/extensions/picocli/deployment/src/main/java/io/quarkus/picocli/deployment/PicocliProcessor.java
@@ -40,8 +40,8 @@ class PicocliProcessor {
 
     @BuildStep
     ModuleEnableNativeAccessBuildItem allowNativeLibraryLoad() {
-        // Need to generate an --enable-native-access because of org.fusesource.jansi.internal.JansiLoader
-        return new ModuleEnableNativeAccessBuildItem("org.fusesource.jansi");
+        // Need to generate an --enable-native-access because of org.jline.nativ.JLineNativeLoader
+        return new ModuleEnableNativeAccessBuildItem("org.jline.jansi");
     }
 
     @BuildStep

--- a/independent-projects/enforcer-rules/src/main/resources/enforcer-rules/quarkus-banned-dependencies.xml
+++ b/independent-projects/enforcer-rules/src/main/resources/enforcer-rules/quarkus-banned-dependencies.xml
@@ -117,6 +117,8 @@
                 <exclude>io.github.dmlloyd:jdk-classfile-backport</exclude>
                 <!-- org.lz4:lz4-java is abandoned and has a CVE -->
                 <exclude>org.lz4:lz4-java</exclude>
+                <!-- org.fusesource.jansi:jansi is abandoned and has a CVE -->
+                <exclude>org.fusesource.jansi:jansi</exclude>
             </excludes>
             <includes>
                 <!-- this is for REST Assured -->

